src/SignalR/server/Core/src/Internal/TypedClientBuilder.cs

```diff
@@ -40,18 +43,20 @@ public static void Validate()
             var moduleBuilder = assemblyBuilder.DefineDynamicModule(ClientModuleName);
             var clientType = GenerateInterfaceImplementation(moduleBuilder);
 
-            return proxy => (T)Activator.CreateInstance(clientType, proxy);
+            var ctor = clientType.GetConstructor(ParameterTypes);
```

Comment:
Can you do this without the expression trees?

Reply:
You mean compile the lambda by hand writing IL? 🤔

Reply:
Delegate.CreateDelegate?

Reply:
Dunno. Can look into it. Are you worried about the compile cost? Or compat.?

Reply:
The dependency on expression trees isn’t great for linkability (it’s a big dependency). We already have dependencies but it would be great to not add new ones

Reply:
I don't see how I can use `Delegate.CreateDelegate` for a ctor without going through `DynamicMethod` and writing the IL manually. Any pointers?

Reply:
Yea I looked, doesn’t look simple. This is fine for now

Reply:
It's not that bad:

```diff
diff --git a/src/SignalR/server/Core/src/Internal/TypedClientBuilder.cs b/src/SignalR/server/Core/src/Internal/TypedClientBuilder.cs
index 8d2599b433..c1cd9c278d 100644
--- a/src/SignalR/server/Core/src/Internal/TypedClientBuilder.cs
+++ b/src/SignalR/server/Core/src/Internal/TypedClientBuilder.cs
@@ -4,7 +4,6 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using System.Linq.Expressions;
 using System.Reflection;
 using System.Reflection.Emit;
 using System.Threading;
@@ -44,10 +43,15 @@ namespace Microsoft.AspNetCore.SignalR.Internal
             var clientType = GenerateInterfaceImplementation(moduleBuilder);
 
             var ctor = clientType.GetConstructor(ParameterTypes);
-            var proxy = Expression.Parameter(typeof(IClientProxy), "proxy");
-            var ctorCall = Expression.New(ctor, proxy);
+            var method = new DynamicMethod("CreateTypedClient", clientType, ParameterTypes, clientType);
 
-            return Expression.Lambda<Func<IClientProxy, T>>(ctorCall, proxy).Compile();
+            var generator = method.GetILGenerator();
+
+            generator.Emit(OpCodes.Ldarg_0);
+            generator.Emit(OpCodes.Newobj, ctor);
+            generator.Emit(OpCodes.Ret);
+
+            return (Func<IClientProxy, T>)method.CreateDelegate(typeof(Func<IClientProxy, T>));
         }
 
         private static Type GenerateInterfaceImplementation(ModuleBuilder moduleBuilder)
```

Which gave the following benchmark:

``` ini
BenchmarkDotNet=v0.10.13, OS=Windows 10.0.17134
Intel Core i7-6820HQ CPU 2.70GHz (Skylake), 1 CPU, 8 logical cores and 4 physical cores
.NET Core SDK=5.0.100-alpha1-013788
  [Host]     : .NET Core 5.0.0-alpha1.19470.7 (CoreCLR 5.0.19.46201, CoreFX 5.0.19.46207), 64bit RyuJIT
  Job-SWFGEY : .NET Core 5.0.0-alpha1.19470.7 (CoreCLR 5.0.19.46201, CoreFX 5.0.19.46207), 64bit RyuJIT

Runtime=Core  Server=True  Toolchain=.NET Core 5.0  
RunStrategy=Throughput  

```
| Method |     Mean |     Error |    StdDev |         Op/s |  Gen 0 | Allocated |
|------- |---------:|----------:|----------:|-------------:|-------:|----------:|
|  Build | 16.14 ns | 0.4467 ns | 0.5649 ns | 61,953,695.7 | 0.0006 |      24 B |

Reply:
How is that slower 😄, noise maybe?

Reply:
Yeah, those 2 ns is probably within the margin of error 😄